rllib/algorithms/ppo/ppo_catalog.py

```diff
@@ -76,30 +76,20 @@ def __init__(
             shared=self.model_config_dict["vf_share_layers"],
         )
 
-        post_fcnet_hiddens = self.model_config_dict["post_fcnet_hiddens"]
-        post_fcnet_activation = self.model_config_dict["post_fcnet_activation"]
+        self.pi_and_vf_head_hiddens = self.model_config_dict["post_fcnet_hiddens"]
+        self.pi_and_vf_head_activation = self.model_config_dict["post_fcnet_activation"]
 
-        pi_head_config_class = (
-            FreeLogStdMLPHeadConfig
-            if self.model_config_dict["free_log_std"]
-            else MLPHeadConfig
-        )
-        self.pi_head_config = pi_head_config_class(
-            input_dims=self.latent_dims,
-            hidden_layer_dims=post_fcnet_hiddens,
-            hidden_layer_activation=post_fcnet_activation,
-            output_activation="linear",
-            # We don't know the output dimension yet, because it depends on the
-            # action distribution input dimension.
-            output_dims=None,
-        )
+        # We don't have the exact (framework specific) action dist class yet and thus
+        # cannot determine the exact number of output nodes (action space) required.
+        # -> Build pi config only in the `self.build_pi_head` method.
+        self.pi_head_config = None
```

Comment:
I think it would be cool if you would pass a framework and then immediately after constructing a catalog, you could build the pi head or modify it's config.
That should feel more natural and would be aligned with https://github.com/ray-project/ray/pull/36057/files#r1217000670

Reply:
Actually, let's do this in a separate PR. This one will get too large and unwieldy otherwise :)